ledger-api/grpc-definitions/com/daml/ledger/api/v2/update_service.proto

```diff
@@ -0,0 +1,127 @@
+import "com/daml/ledger/api/v2/reassignment.proto";
+
+option java_outer_classname = "UpdateServiceOuterClass";
+option java_package = "com.daml.ledger.api.v2";
+option csharp_namespace = "Com.Daml.Ledger.Api.V2";
+
+// TODO add documentation of participant offsets and virtual shared ledger and causality guarantees
+// Allows clients to read updates (transactions and reassignments) from the ledger.
+service UpdateService {
+
+  // Read the ledger's filtered transaction stream and related reassignments for a set of parties.
+  // For transactions it lists only creates and archives, but not other events.
+  // Omits all events on transient contracts, i.e., contracts that were both created and archived in the same transaction.
+  rpc GetUpdates (GetUpdatesRequest) returns (stream GetUpdatesResponse);
+
+  // Read the ledger's complete transaction tree stream and related reassignments for a set of parties.
+  // The stream will be filtered only by the parties as wildcard parties.
+  // The template/interface filters describe the respective fields in the ``CreatedEvent`` results.
+  rpc GetUpdateTrees (GetUpdatesRequest) returns (stream GetUpdateTreesResponse);
+
+  // Lookup a transaction tree by the ID of an event that appears within it.
+  // For looking up a transaction instead of a transaction tree, please see GetFlatTransactionByEventId
+  rpc GetTransactionByEventId (GetTransactionByEventIdRequest) returns (GetTransactionResponse);
+
+  // Lookup a transaction tree by its ID.
+  // For looking up a transaction instead of a transaction tree, please see GetFlatTransactionById
+  rpc GetTransactionById (GetTransactionByIdRequest) returns (GetTransactionResponse);
+
+  // Lookup a transaction by the ID of an event that appears within it.
+  rpc GetFlatTransactionByEventId (GetTransactionByEventIdRequest) returns (GetFlatTransactionResponse);
+
+  // Lookup a transaction by its ID.
+  rpc GetFlatTransactionById (GetTransactionByIdRequest) returns (GetFlatTransactionResponse);
+}
+
+message GetUpdatesRequest {
+  // Beginning of the requested ledger section.
+  // This offset is exclusive: the response will only contain transactions whose offset is strictly greater than this.
+  // Required
+  com.daml.ledger.api.v1.LedgerOffset begin = 1;
+
+  // End of the requested ledger section.
+  // This offset is inclusive: the response will only contain transactions whose offset is less than or equal to this.
+  // Optional, if not set, the stream will not terminate.
+  com.daml.ledger.api.v1.LedgerOffset end = 2;
+
+  // Requesting parties with template filters.
+  // Template filters must be empty for GetUpdateTrees requests.
+  // Required
+  TransactionFilter filter = 3;
+
+  // If enabled, values served over the API will contain more information than strictly necessary to interpret the data.
+  // In particular, setting the verbose flag to true triggers the ledger to include labels for record fields.
```

Comment:
it's more than labels, it's also record and variant type ids

Reply:
added, tanks